pkg/authorization/api/types.go

```diff
@@ -75,6 +75,8 @@ type PolicyRule struct {
 	AttributeRestrictions kruntime.EmbeddedObject `json:"attributeRestrictions"`
 	// Resources is a list of resources this rule applies to.  ResourceAll represents all resources.
 	Resources []string `json:"resources"`
+	// ResourceNames is an optional white list of names that the rule applies to.  An empty set means that everything is allowed.
+	ResourceNames kutil.StringSet
```

Comment:
> any reason we can't use all []string or StringSet?

Task for a different pull?  I do want to do it, but it wasn't related to this change.